### DISCUSSION TITLE
Controller ref changes with every render, forcing input re-render

### DISCUSSION THREAD

**Discussion by trevorr** [react-hook-form/react-hook-form#10549] [General]
I'm noticing an issue with unnecessary re-rendering because it seems that the `ref` passed to `Controller`'s `render` function is changing on every `Controller` render. In the [code for `useController`](https://github.com/react-hook-form/react-hook-form/blob/master/src/useController.ts), I see that [`onChange`](https://github.com/react-hook-form/react-hook-form/blob/master/src/useController.ts#L118) and [`onBlur`](https://github.com/react-hook-form/react-hook-form/blob/master/src/useController.ts#L129) are wrapped in `React.useCallback` but [`ref`](https://github.com/react-hook-form/react-hook-form/blob/master/src/useController.ts#L140) is not. Is this an oversight, or is there a reason for it?

In my case, the re-rendering is not (just) a performance issue. The re-rendered component in question is a [Lexical editor](https://lexical.dev/)/content-editable, which tends to steal focus on render. I've worked around it on initial render by restoring focus, but that's not feasible other times because the focus jump would be noticeable. My current workaround is to store the ref in a ref, so the most recent ref gets called when the DOM element changes, but changing the ref doesn't force a re-render. However, this makes the possibly invalid assumption that the old ref function is still correctly bound to the form.

```ts
<Controller
  control={formMethods.control}
  name="description"
  render={({ field: { name, onChange, value, ref }, fieldState: { error } }) => (
    <LexicalTextField
      id={name}
      value={value}
      onChange={onChange}
      error={!!error}
      helperText={error?.message}
      inputRef={ref} // <-- ref changes on every render, forcing LexicalTextField to re-render its content-editable
    />
  )}
/>

```

**Comment by leapful** [comment]
@trevorr 

The reason of this implementation came from this PR : https://github.com/react-hook-form/react-hook-form/pull/9130 as a workaround for handling `field.ref` in React strict mode.

You can apply `React.memo` as a wrapper component for your target form field element then apply custom equal condition for the memorized wrapper to avoid `field.ref` is changed on each render.

For example: https://codesandbox.io/s/react-hook-form-v7-controller-ts-forked-r84gpp?file=/src/index.tsx
